File: 29-01-2024/Questions/Cpp Solution/Solution3.cpp
#include<stdio.h>
#include<string.h>

int main(){
	char teams_list[12][20],str[20];
	int count=0,i,j;
	scanf("%s\n",str);
	while(strcmp(str,"q")!=0&&strcmp(str,"Q")!=0){
		if(count==12){
			printf("Maximum limit of Teams is 12");
			return 0;
		}
		strcpy(teams_list[count],str);
		count+=1;
		scanf("%s\n",str);
	}
	if(count>=3){
		printf("TOTAL MATCHES:%d\n",(count)*(count-1)/2);
		for(i=0;i<count-1;i++){
			for(j=i+1;j<count;j++){
				printf("%s-VS-%s\n",teams_list[i],teams_list[j]);
			}
		}
	}
	else{
		printf("Minimum 3 Teams are required");
	}
}

File: 29-01-2024/Questions/Python Solution/Solution3.py
def Indian_Premier_League(s, teams_list):

    count = 0
    while s != 'q' and s != 'Q':
        if count == 12:
            print('Maximum limit of Teams is 12')
            break

        teams_list.append(s)
        count += 1
        s = input()

        if count >= 3:
            n = (count-1) * (count) // 2    
            print('TOTAL MATCHES: ',n)

            for i in range(count - 1):
                for j in range(i + 1, count):
                    print(teams_list[i] + '-VS-' + teams_list[j])

        else:
            print('Minimum 3 Teams are required')


s = input()
teams_list = []

out_ = Indian_Premier_League(s, teams_list)
print(out_)
